image/tools/lib/component/redis.sh

```diff
@@ -0,0 +1,15 @@
+function component_dump_data {
+    dest_file="$1/dump.rdb"
+    dumb_rdb_path="/var/lib/redis/data/dump.rdb"
+
+    oc projects
+    redis_pod_name=$(oc get pods -l deploymentConfig=backend-redis -o name -n 3scale | sed -e 's/^pod\///')
+    copy_output=$(oc cp 3scale/${redis_pod_name}:${dumb_rdb_path} ${dest_file})
+    echo ${copy_output}
+    
+    # check if rdb file was rewritten during oc cp, and copy it again if it was
+    if [[ $copy_output == *"file changed as we read it"* ]]; then
```

Comment:
It's a bit hard to replicate this scenario because .rdb file is very small on a fresh 3Scale instance.
But I did test it, screenshot below. And if necessary, I can walk a verifyer through it.

![screenshot 2019-03-01 at 16 42 25](https://user-images.githubusercontent.com/1605799/53652492-b0ef4880-3c51-11e9-8003-c0e93192b77f.png)

Reply:
I am creating a util function for this (https://github.com/integr8ly/backup-container-image/pull/6#discussion_r261592156), so I will incorporate this check there too.
I will try retry a number of times there